File: agent-service/schemas/workflow.py
from typing import List, Optional, Literal
from pydantic import BaseModel
class WorkflowNode(BaseModel):
    id: str
    type: Literal['agent','team','tool','human_approval','condition','summary']
    label: str
    agentId: Optional[str] = None
    requireApproval: bool = False
class WorkflowEdge(BaseModel):
    from_: str
    to: str
    condition: Optional[str] = None
class WorkflowConfig(BaseModel):
    id: str
    name: str
    version: str = '1.0.0'
    nodes: List[WorkflowNode]
    edges: List[WorkflowEdge] = []
    failureStrategy: str = 'ask_user'
    confirmPolicy: str = 'all'

File: src/types/workflow.ts
export type WorkflowNodeType = 'agent' | 'human_approval' | 'condition' | 'tool' | 'summary';

export interface WorkflowNode {
  id: string;
  name: string;
  type: WorkflowNodeType;
  agentId?: string;
  inputFields: string[];
  outputFields: string[];
  onFailure: string;
  maxRetries: number;
  timeoutSeconds: number;
  enabled: boolean;
}

export interface WorkflowConfig {
  id: string;
  name: string;
  description: string;
  type: string;
  failureStrategy: string;
  retryLimit: number;
  nodeTimeoutSeconds: number;
  confirmPolicy: string;
  jsonVersion: number;
  default: boolean;
  enabled: boolean;
  nodes: WorkflowNode[];
  jsonPreview?: string;
}
